src/raftstore/store/fsm/peer.rs

```diff
@@ -567,7 +583,15 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         if let Some(p) = self.fsm.peer.take_apply_proposals() {
             proposals.push(p);
         }
-        self.fsm.peer.handle_raft_ready_append(self.ctx);
+        let res = self.fsm.peer.handle_raft_ready_append(self.ctx);
+        if let Some(r) = res {
+            self.on_role_changed(&r.0);
```

Comment:
Looks like it has been handled in `handle_raft_ready_append ` already?

Reply:
Different layers execute different commands.